File: ROS_topic/src/topic_pkg/src/sub_node.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

void callbackFunction(const std_msgs::String::ConstPtr& msg) {
  ROS_INFO("I heard: %s", msg->data.c_str());
}

int main(int argc, char **argv) {
  ros::init(argc, argv, "sub_node_name");
  ros::NodeHandle nh;
  ros::Subscriber sub = nh.subscribe("topic_name", 10, callbackFunction);

  ros::spin();  // 设置循环调用回调函数

  return 0;
}

File: catkin_ws/src/topic_pkg/scripts/sub_node.py
#! /usr/bin/env python3
#coding:utf-8
import rospy
from std_msgs.msg import String

def CallBack_Function(msg):
    rospy.loginfo("I heard:%s",msg.data)

if __name__ == "__main__":
    # 初始化 ROS 节点:命名(唯一)
    rospy.init_node("sub_node_name")
    # 实例化 订阅者 对象
    sub = rospy.Subscriber("topic_name",String,CallBack_Function,queue_size=10)
    # CallBack_Function 处理订阅的消息(回调函数) 
    # 只用填函数名，ros会自动将订阅到的消息作为其入口参数传给回调函数
    # 设置循环调用回调函数
    rospy.spin()
